include/cbl/CBLPlatform.h

```diff
@@ -0,0 +1,53 @@
+//
+// CBLPlatform.h
+//
+// Copyright © 2019 Couchbase. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#pragma once
+#include "CBLBase.h"
+
+CBL_CAPI_BEGIN
+
+#ifdef __ANDROID__
+
+/** \defgroup android  Android
+     @{ */
+
+/** Application context information required for Android application to initialize before using
+    CouchbaseLite library. */
+typedef struct {
+    const char* filesDir;   ///< The directory where the opened database will be stored when a specific database directory is not specified in \ref CBLDatabaseConfiguration.
+                            ///< Normally the path returned from Android Context's getFilesDir() method can be specified here unless different directory is desired.
+                            ///< The specified fileDir must exist otherwise an error will be returend when calling \r CBL_Init().
+    
+    const char* tempDir;    ///< The directory where the SQLite stores its temporary files.
+                            ///< Normally the path returned from Android Context's getExternalFilesDir(String type) with a custom type such as "CBLTemp" can be specified here
+                            ///< unless different directory is desired. The specified tempDir must exist otherwise an error will be returend when calling \r CBL_Init().
```

Comment:
I believe the temp directory will be automatically set based on the `filesDir`.  I checked the Java implementation and it seems to do so.  This must be on the same partition as the filesDir so it is probably safer to calculate this based on `filesDir` rather than giving free reign to specify it.

Reply:
Based on https://github.com/couchbase/couchbase-lite-java-common/blob/master/android/main/java/com/couchbase/lite/CouchbaseLite.java#L39, the current implementation in Java doesn't derive the temp directory based on the filesDir.

> This must be on the same partition as the filesDir 
Is there a reason that this needs to be baed on the same partition as filesDir? IIRC, the same partition requirement

Reply:
Actually it is not this temp directory that has this requirement, it is the c4 temp directory.  I forgot there were two temp directories....

Reply:
I'm creating CBL-2241.